content/03-reference/01-tools-and-interfaces/02-prisma-client/078-error-reference.mdx

```diff
@@ -0,0 +1,92 @@
+| `P3002`   | "The attempted migration was rolled back: \${database_error}"<br />                                                                                                              |
+| `P3003`   | "The format of migrations changed, the saved migrations are no longer valid. To solve this problem, please follow the steps at: https://pris.ly/d/migrate#troubleshooting"<br /> |
+| `P3004`   | "The `${database_name}` database is a system database, it should not be altered with prisma migrate. Please connect to another database."<br />                                  |
+
+#### Common
+
+| **Error** | **Example message**                                                                                                                                                                                                                                                                                                                                                                                                                                                     | **Description** |
+| :-------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :-------------- |
+| `P1000`   | "Authentication failed against database server at `${database_host}`, the provided database credentials for `${database_user}` are not valid. Please make sure to provide valid database credentials for the database server at `${database_host}`."<br />                                                                                                                                                                                                              |
+| `P1001`   | "Can't reach database server at `${database_host}`:`${database_port}` Please make sure your database server is running at `${database_host}`:`${database_port}`."<br />                                                                                                                                                                                                                                                                                                 |
+| `P1002`   | "The database server at `${database_host}`:`${database_port}` was reached but timed out. Please try again. Please make sure your database server is running at `${database_host}`:`${database_port}`. "<br />"Database `${database_name}.${database_schema_name}` does not exist on the database server at `${database_host}:${database_port}`."<br />"Database `${database_name}` does not exist on the database server at `${database_host}:${database_port}`."<br /> |
```

Comment:
Are the second and third error messages shown with P1002? Or P1003? 

I'm not familiar enough with Rust to know 😅

Reply:
Another good catch - script updated, those 3 messages now belong to `P1003` as they should. 💚